File: apps/api/src/modules/hello/dto/hello-response.dto.ts
import { ApiProperty } from '@nestjs/swagger';

/** Response shape for GET /api/hello */
export class HelloResponseDto {
  @ApiProperty({
    description: 'Greeting message from the API',
    example: 'Hello from NestJS',
  })
  message!: string;

  @ApiProperty({
    description: 'ISO 8601 timestamp of when the response was generated',
    example: '2026-04-26T22:30:00.000Z',
  })
  timestamp!: string;
}

File: apps/mobile/lib/features/hello/data/dto/hello_response_dto.dart
/// Placeholder DTO for the backend `GET /api/hello` response.
///
/// Phase 06 replaces this with the generated model from `packages/api_client`
/// (OpenAPI-generated via `dio` generator). At that point this file will be
/// deleted and the import in `hello_repository_impl.dart` will be updated.
///
/// Response envelope expected from the API:
/// ```json
/// {
///   "data": {
///     "message": "Hello from NestJS!",
///     "timestamp": "2026-04-26T12:00:00.000Z"
///   },
///   "meta": {},
///   "requestId": "abc-123"
/// }
/// ```
class HelloResponseDto {
  const HelloResponseDto({
    required this.message,
    required this.timestamp,
  });

  factory HelloResponseDto.fromJson(Map<String, dynamic> json) =>
      HelloResponseDto(
        message: json['message'] as String,
        timestamp: json['timestamp'] as String,
      );

  final String message;
  final String timestamp;

  @override
  String toString() =>
      'HelloResponseDto(message: $message, timestamp: $timestamp)';
}
